runtime/parachains/src/paras/mod.rs

```diff
@@ -2127,7 +2127,7 @@ impl<T: Config> Pallet<T> {
 /// or removing parachains in bulk.
 pub(crate) struct ParachainsCache<T: Config> {
 	// `None` here means the parachains list has not been accessed yet, nevermind modified.
-	parachains: Option<Vec<ParaId>>,
+	parachains: Option<BTreeSet<ParaId>>,
```

Comment:
Does this 100% have the same SCALE encoding format as the `Vec`? If not, this will break things, badly.